src/pydap/model.py

```diff
@@ -393,32 +384,25 @@ def __getattr__(self, attr):
             return DapType.__getattr__(self, attr)
 
     def __contains__(self, key):
-        return (key in self.visible_keys)
+        return (key in self._visible_keys)
 
     # __iter__, __getitem__, __len__ are required for Mapping
     # From these, keys, items, values, get, __eq__,
     # and __ne__ are obtained.
     def __iter__(self):
         for key in self._dict.keys():
-            if key in self.visible_keys:
+            if key in self._visible_keys:
                 yield key
 
-    def all_keys(self):
+    def _all_keys(self):
```

Comment:
In fact ``_all_keys`` is the only required method. I've deleted ``_all_items`` and ``_all_values``.